tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs

```diff
@@ -428,6 +429,171 @@ public async Task PersistenceIsDisabledByDefault()
         }
     }
 
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    [RequiresDocker]
+    public async Task RedisInsightWithDataShouldPersistStateBetweenUsages(bool useVolume)
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+
+        string? volumeName = null;
+        string? bindMountPath = null;
+
+        try
+        {
+            using var builder1 = TestDistributedApplicationBuilder.CreateWithTestContainerRegistry(testOutputHelper);
+            IResourceBuilder<RedisInsightResource>? redisInsightBuilder1 = null;
+            var redis1 = builder1.AddRedis("redis")
+                .WithRedisInsight(c => { redisInsightBuilder1 = c; });
+            Assert.NotNull(redisInsightBuilder1);
+
+            if (useVolume)
+            {
```

Comment:
Would the test look nicer if we moved the `WithDataVolume` call into the `WithRedisInsight` callback instead? That way we didn't have this variable capturing.

Reply:
variable capturing is necessary for another part of the test.
see:
https://github.com/dotnet/aspire/pull/6432/files#diff-3b0aa8b65b60aa9db2aff71ccdede3578734eccad1ca79454241c28b3ecc73b3R556-R569